django/contrib/admin/views/autocomplete.py

```diff
@@ -24,11 +26,29 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
+        app_label = request.GET.get('app_label')
+        model_name = request.GET.get('model_name')
+        field_name = request.GET.get('field_name')
+
+        model = apps.get_model(app_label, model_name)
+
+        try:
+            self.field = model._meta.get_field(field_name)
+        except FieldDoesNotExist:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
+
+        if self.field.remote_field is None or self.field.remote_field.model != self.model_admin.model:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
+
+        to_field_name = getattr(self.field.remote_field, 'field_name', self.model_admin.model._meta.pk.name)
```

Comment:
From reading through Django's source code, you can rely that `self.field_remote_field.field_name` is set I think:

https://github.com/django/django/blob/a8b3f96f6acfa082f99166e0a1cfb4b0fbc0eace/django/db/models/fields/related.py#L945-L948

Reply:
This code is executed only for `ForeignKey`s not M2M or any other custom `RelatedObject` a developer might create.